docs/DeveloperGuide.md

```diff
@@ -6,33 +6,360 @@
+- The first substring is a one-character string that represents the appointment type the UID is generated to.  
+  For example, an appointment type of `Surgery` will be denoted as `"S"`.
+- The second substring is an eight character long string with each character representing a digit from 0-9.  
+  Each digit is chosen at random by using Java's inbuilt `Random` class. An example of this substring 
+  will be `"01234567"`.
+- The third substring is a one-character string representing a random letter chosen from A-Z.  
+  This letter is also chosen using Java's inbuilt `Random` class.
+- The UID string is concatenated together using Java's inbuilt `StringBuilder` class. The `StringBuilder` class provides
+  a faster way to concatenate strings as compared to using the `+` operator.   
+  An example of a randomly generated UID string for each appointment type is shown below:
+  - Consultation: `C82739812B`
+  - Vaccination: `V71829748S`
+  - Surgery: `S23847989T`
+
+
+###### Alternatives
+Other alternatives that was considered are:
+- Using the Java inbuilt UUID class.
+   - Pros: Do not need to implement the class itself. Easy to use.
+   - Cons: Overkill as it generates a 128 bit String. Cannot customise to fit our custom uid format.
+- Asking the user to manually key in a UID string themselves.
+   - Pros: Do not need to implement the feature itself.
+   - Cons: We felt that this is counter-intuitive as it increases workload and human error. Automating
+     this process will reduce potential errors.
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+Veterinarians in Vet Clinics around Singapore who
+* prefer typing over using a mouse.
+* need to keep track a large number of appointments.
+* prefer Command Line Interface(CLI) over other interfaces.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+Veterinary Clinics have a large number of appointments for the veterinarians and their admin staff to handle.
+
+Sniff is a
+appointment manager that helps clinics keep track of their appointments. This eases the workload of the clinic staff
+and helps improve efficiency in running a Vet clinic, while reducing human errors occurring in the workplace.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+
+| Version | As a ... | I want to ...                               | So that I can ...                                        |
+|---------|----------|---------------------------------------------|----------------------------------------------------------|
+| v1.0    |new user| add new appointments                        |                                                          |
```

Comment:
I understand how this is self-explanatory, but it might be good to rephrase/ go more in-depth in the "so-that-I-can" .
(e.g. I want to add new appointments so that I can be able to easily book appointments with a doctor).